### ISSUE TITLE
Runtime interface in 1.2

### ISSUE THREAD

**Issue by nityarai** [ddemidov/amgcl#124]
Two query

1. How can we use runtime interface in released version 1.2. The example in the does not work with 1.2 release.
2. When do you plan to release the next stable version of the code.

**Comment by ddemidov** [comment]
Something like this should work in 1.2:
```cpp
#include <amgcl/adapter/crs_tuple.hpp>
#include <amgcl/backend/builtin.hpp>
#include <amgcl/runtime.hpp>
#include <amgcl/make_solver.hpp>

int main() {
    typedef amgcl::backend::builtin<double> Backend;
    typedef amgcl::make_solver<
        amgcl::runtime::amg<Backend>,
        amgcl::runtime::iterative_solver<Backend>
        > Solver;

    int n;
    std::vector<int> ptr, col;
    std::vector<double> val;

    Solver solve(boost::tie(n, ptr, col, val));
}
```

There were some breaking changes with sphinx/readthedocs, do they are not able to render 1.2 docs anymore, so I have attached a copy of 1.2 html documentation to the 1.2 release here:
https://github.com/ddemidov/amgcl/releases/tag/1.2.0.

Re next release, the master branch is pretty stable at this point and has a lot of exciting changes, so I should recommend using it instead. I still need to find time to update the documentation to do the next release though, so I think I'll do a pre-release instead in the next couple of days.

**Comment by ddemidov** [comment]
https://github.com/ddemidov/amgcl/releases/tag/1.3.99

**Comment by nityarai** [comment]
Thanks Denis, that works. I see you have blaze interface supported as well, i could not find an example illustrating the use of blaze CSR matrix with the solver. Could you help with the example for that.

**Comment by ddemidov** [comment]
Just include `<amgcl/backend/blaze.hpp>` and use blaze matrices as input to amgcl constructor.
There is a unified test here: https://github.com/ddemidov/amgcl/blob/master/tests/test_solver_blaze.cpp, and you can build examples/solver.cpp for the blaze backend (make solver_blaze).

**Comment by nityarai** [comment]
In 1.2 version i do

#include "amgcl/backend/blaze.hpp"
#include <blaze/math/CompressedMatrix.h>
blaze::DynamicVector<value_type>;

using Backend = amgcl::backend::blaze<value_type>;
using Solver =  amgcl::make_solver<amgcl::runtime::amg<Backend>, amgcl::runtime::iterative_solver<Backend>>;

using Matrix =  BlazeMatrixType<value_type>;
using Vector =  BlazeVectorType<value_type>;

Matrix &m = loadMatrix();
Solver solver_ = Solver(m, parameters_);

and compiler complains

----
error: no type named ‘ROW_ITERATOR_NOT_IMPLEMENTED’ in ‘class blaze::CompressedMatrix<double, false>’
     typedef typename Matrix::ROW_ITERATOR_NOT_IMPLEMENTED type;
----

**Comment by ddemidov** [comment]
This should be fixed by 255fb02a2a2ca4350eafb7b4717111066898e6b6 (which, of course, you would not be able to use if you insist on using v1.2).
